### PR TITLE
✨ add Maingame

### PR DESCRIPTION
**Pull request by LouisaStrunk02**
Changes:

- gameplay created
- controls the validity of the moves

### CODE REVIEW

File: Board.py [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
-            [placeholder] * 7,
-            [placeholder] * 7,
-            [placeholder] * 7,
+            [self.placeholder] * 7,
+            [self.placeholder] * 7,
+            [self.placeholder] * 7,
+            [self.placeholder] * 7,
+            [self.placeholder] * 7,
+            [self.placeholder] * 7,
             ["1", "2", "3", "4", "5", "6", "7"]
```
Comment (stefanluth) [inline Board.py:14]:
```suggestion
            ["1", "2", "3", "4", "5", "6", "7"],
```

File: Board.py [hunk 2]
```diff
@@ -21,4 +20,23 @@ def show_board(self) -> None:
         for line in self.board:
             for element in line:
                 print("|" + element, end="")
+
```
Comment (stefanluth) [inline Board.py:23]:
```suggestion

```

File: MainGame.py [hunk 3]
```diff
@@ -3,22 +3,62 @@
...
+        self.turn_number = 0
 
     def get_user_name(self, salutation: str) -> str:
         while True:
-            user_name = input(f"{salutation}: Please enter your name: ")
-
-            input_is_valid = user_name.isalpha() and len(user_name) > 0
+            user_name: str = input(f"{salutation}: Please enter your name: ")
+            input_is_valid: bool = user_name.isalpha() and len(user_name) > 0
+
```
Comment (stefanluth) [inline MainGame.py:15]:
```suggestion

```

File: MainGame.py [hunk 4]
```diff
@@ -3,22 +3,62 @@
...
+                return
+            else:
+                print("You can't put a coin in column " + str(column) + ". Try again.")
+
+    def get_column_input(self, player_name: str) -> int:
+        while True:
+            try:
+                column_input = int(input(player_name + ", select a column to put your coin: "))
+                input_is_valid = 1 <= column_input <= 7
+
```
Comment (stefanluth) [inline MainGame.py:46]:
```suggestion

```

File: MainGame.py [hunk 5]
```diff
@@ -3,22 +3,62 @@
...
+                if input_is_valid:
+                    return column_input
+                else:
+                    print("Invalid input. Please try again")
+            except ValueError:
+                print("Please enter an integer.")
+
+    def round(self) -> None:
+        self.turn_number += 1
+
```
Comment (stefanluth) [inline MainGame.py:56]:
```suggestion

```

**Review by stefanluth — CHANGES_REQUESTED** [review]

**Review by DevBastiS — APPROVED** [review]
What Stefan said

**Pull request merged**